test/lifecycle/DestructibleDelayed.test.js

```diff
@@ -0,0 +1,127 @@
+require('../helpers/transactionMined.js');
+
+require('chai')
+  .use(require('chai-as-promised'))
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+const twoWeeks = 1209600;
+
+contract('DestructibleDelayed', function ([owner, anyone]) {
+  let destructibleDelayed;
+
+  describe('Destruction requests', function () {
+    beforeEach(async function () {
+      destructibleDelayed = await DestructibleDelayedMock.new(
+        twoWeeks, { from: owner, value: web3.toWei('10', 'ether') }
+      );
+    });
+
+    it('owner should be able to create a destroyRequest', async function () {
+      await expectEvent.inTransaction(
+        destructibleDelayed.destroyRequest({ from: owner }),
+        'SelfDestructionRequest'
+      );
+
+      const destructionRequested = await destructibleDelayed.destructionRequested();
+
+      destructionRequested.should.be.equal(true);
+    });
+
+    it('anyone should not be able to create a destroy request', async function () {
+      await expectThrow(
+        destructibleDelayed.destroyRequest({ from: anyone })
+      );
+    });
+    
+    describe('once destruction requested,', function () {
+      var block;
+      var tx;
+
+      beforeEach(async function () {
+        tx = await destructibleDelayed.destroyRequest({ from: owner });
+        block = await web3.eth.getBlock(tx.receipt.blockNumber);
+      });
+
+      it('destruction time should be set to block.timestamp + destruction_delay', async function () {
+        const destructionTime = await destructibleDelayed.destructionTime();
+        const destructionDelay = await destructibleDelayed.SELFDESTRUCTION_DELAY();
+
+        destructionTime.should.be.bignumber.equal(
+          destructionDelay.plus(block.timestamp)
+        );
+      });
+
+      it('should not allow a new #destroyRequest call', async function () {
+        await expectThrow(
+          destructibleDelayed.destroyRequest({ from: owner })
+        );
+      });
+
+      it('destroyRequest can be cancelled by owner', async function () {
+        await expectEvent.inTransaction(
+          destructibleDelayed.cancelDestroyRequest({ from: owner }),
+          'SelfDestructionRequestCancelled'
+        );
+
+        const destructionRequested = await destructibleDelayed.destructionRequested();
+
+        destructionRequested.should.be.equal(false);
+      });
+
+      it('anyone should not be able to cancel a destroy request', async function () {
+        await expectThrow(
+          destructibleDelayed.cancelDestroyRequest({ from: anyone })
+        );
+      });
+
+      it('should not allow to call #destroy before delay is passed', async function () {
+        await expectThrow(
+          destructibleDelayed.destroyRequest({ from: owner })
+        );
+      });
+
+      describe('once destructionTime is passed', function () {
+        beforeEach(async function () {
+          await web3.currentProvider.send({
```

Comment:
can you use the `increaseTimeTo` helper here?

Reply:
hmm? You should be able to do

```
import { increaseTimeTo } from '../helpers/increaseTime';

...

await increaseTimeTo(twoWeeks + 1);
```

unless I'm mistaken

Reply:
Well, the time needs to be increase *by* `twoWeeks + 1` to guarantee that `destructionTime` is passed. So I either use `increaseTimeTo(destructionTime + 1)`, where I need to do an await call to obtain `destructionTime` (which is just `now + 2 weeks`), or use `increaseTime(twoWeeks + 1`. The latter seems like redoing an operation, since we already know by how much we need to increase the time by. 

I think using this helper function would make the whole thing simpler ; 

`import increaseTime from './helpers/increaseTime';`